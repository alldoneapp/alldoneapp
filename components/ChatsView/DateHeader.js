import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import PropTypes from 'prop-types'
import styles, { colors } from '../styles/global'
import moment from 'moment'
import { getDateFormat } from '../UIComponents/FloatModals/DateFormatPickerModal'
import { translate } from '../../i18n/TranslationService'

function DateHeader({ dateText, isToday, isOverdue, date }) {
    const weekdays = [
        translate('Monday'),
        translate('Tuesday'),
        translate('Wednesday'),
        translate('Thursday'),
        translate('Friday'),
        translate('Saturday'),
        translate('Sunday'),
    ]
    const text = () => {
        let dayName = ''
        let upperCaseDateText = dateText.toUpperCase()

        if (date._isValid) {
            dayName = weekdays[moment(date).isoWeekday() - 1].toUpperCase()
            if (
                upperCaseDateText !== 'TODAY' &&
                upperCaseDateText !== 'TOMORROW' &&
                upperCaseDateText !== 'YESTERDAY'
            ) {
                upperCaseDateText = date.format(getDateFormat())
            } else {
                upperCaseDateText = translate(upperCaseDateText)
            }
        }
        return `${upperCaseDateText} â€¢ ${dayName}`
    }

    return (
        <View
            style={[
                localStyles.container,
                isToday && localStyles.containerToday,
                isOverdue && localStyles.overdueContainer,
            ]}
        >
            <View style={[localStyles.innerContainer, isOverdue && localStyles.overdueIContainer]}>
                <Text style={[styles.overline, isOverdue ? localStyles.overdueText : localStyles.dateText]}>
                    {text()}
                </Text>
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        paddingTop: 24,
        paddingBottom: 8,
    },
    containerToday: {
        paddingTop: 8,
        paddingBottom: 8,
    },
    innerContainer: {
        flexDirection: 'row',
        backgroundColor: colors.Grey100,
        borderRadius: 4,
        height: 24,
        alignItems: 'center',
    },
    overdueContainer: {
        paddingTop: 26,
        paddingBottom: 2,
    },
    overdueIContainer: {
        height: 20,
        backgroundColor: '#FFFFFF',
    },
    dateText: {
        color: colors.Text02,
        zIndex: 1,
        paddingLeft: 12,
    },
    overdueText: {
        ...styles.body3,
        letterSpacing: 1.5,
        color: colors.Text03,
        textTransform: 'uppercase',
    },
})

DateHeader.propTypes = {
    dateText: PropTypes.string,
    isToday: PropTypes.bool,
    isOverdue: PropTypes.bool,
}

DateHeader.defaultProps = {
    dateText: 'Today',
    isToday: false,
    isOverdue: false,
}

export default DateHeader
