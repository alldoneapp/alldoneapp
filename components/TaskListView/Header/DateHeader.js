import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import moment from 'moment'

import styles, { colors } from '../../styles/global'
import { getDateFormat } from '../../UIComponents/FloatModals/DateFormatPickerModal'
import { translate } from '../../../i18n/TranslationService'
import { generateDateHeaderText } from '../../../utils/EstimationHelper'

export default function DateHeader({
    dateText,
    isToday,
    isOverdue,
    estimation,
    date,
    firstDateSection,
    amountTasks,
    projectId,
}) {
    const weekdays = [
        translate('Monday'),
        translate('Tuesday'),
        translate('Wednesday'),
        translate('Thursday'),
        translate('Friday'),
        translate('Saturday'),
        translate('Sunday'),
    ]

    let dayName = ''
    let upperCaseDateText = dateText.toUpperCase()
    if (date._isValid) {
        dayName = weekdays[moment(date).isoWeekday() - 1].toUpperCase()
        if (upperCaseDateText !== 'TODAY' && upperCaseDateText !== 'TOMORROW' && upperCaseDateText !== 'YESTERDAY') {
            upperCaseDateText = date.format(getDateFormat())
        } else {
            upperCaseDateText = translate(upperCaseDateText)
        }
    }

    const text = generateDateHeaderText(projectId, upperCaseDateText, dayName, estimation, amountTasks)

    return (
        <View
            style={[
                localStyles.container,
                isToday || firstDateSection ? localStyles.containerToday : undefined,
                isOverdue && localStyles.overdueContainer,
            ]}
        >
            <View style={[localStyles.innerContainer, isOverdue && localStyles.overdueIContainer]}>
                <View style={{ flex: 1, justifyContent: 'flex-start', flexDirection: 'row' }}>
                    <Text style={[styles.overline, isOverdue ? localStyles.overdueText : localStyles.dateText]}>
                        {text}
                    </Text>
                </View>
            </View>
        </View>
    )
}

DateHeader.defaultProps = {
    dateText: 'Today',
    isToday: false,
    isOverdue: false,
    amountTasks: 0,
}

const localStyles = StyleSheet.create({
    container: {
        paddingTop: 24,
        paddingBottom: 8,
    },
    containerToday: {
        paddingTop: 8,
        paddingBottom: 8,
    },
    innerContainer: {
        flex: 1,
        justifyContent: 'space-between',
        flexDirection: 'row',
        backgroundColor: colors.Grey100,
        borderRadius: 4,
        height: 24,
        alignItems: 'center',
    },
    overdueContainer: {
        paddingTop: 26,
        paddingBottom: 2,
    },
    overdueIContainer: {
        height: 20,
        backgroundColor: '#FFFFFF',
    },
    dateText: {
        color: colors.Text02,
        zIndex: 1,
        paddingLeft: 12,
    },
    overdueText: {
        ...styles.body3,
        letterSpacing: 1.5,
        color: colors.Text03,
        textTransform: 'uppercase',
    },
})
