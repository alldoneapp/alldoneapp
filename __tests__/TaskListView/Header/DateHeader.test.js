import React from 'react'
import DateHeader from '../../../components/TaskListView/Header/DateHeader'

import renderer from 'react-test-renderer'

describe('DateHeader component', () => {
    //Uncomment the the following lines if you need to add more tests in the describe
    //block. React Native Animated needs its internal timers mocked. See: https://github.com/facebook/jest/issues/6434
    //beforeEach(() => {
    jest.useFakeTimers()
    //});

    describe('DateHeader with text snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<DateHeader dateText="TODAY" date={Date.now()} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('Function animateProgress snapshot test', () => {
        it('should execute and render correctly', () => {
            const tree = renderer.create(<DateHeader dateText="TODAY" date={Date.now()}/>)

            tree.getInstance().animateProgress(100)
            expect(tree.toJSON()).toMatchSnapshot()
        })
    })
})
