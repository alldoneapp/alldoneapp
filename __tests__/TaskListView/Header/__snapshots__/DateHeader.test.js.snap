// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader component DateHeader with text snapshot test should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 8,
        "paddingTop": 24,
      },
      undefined,
      false,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFBFB",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 24,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#EBF5FF",
          "borderRadius": 4,
          "height": 24,
          "position": "absolute",
          "width": "0%",
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 11,
            "letterSpacing": 1.5,
            "lineHeight": 20,
          },
          Object {
            "color": "#4E5D78",
            "paddingLeft": 12,
            "zIndex": 1,
          },
        ]
      }
    >
      TODAY • 
    </Text>
  </View>
</View>
`;

exports[`DateHeader component Function animateProgress snapshot test should execute and render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 8,
        "paddingTop": 24,
      },
      undefined,
      false,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFBFB",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 24,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#EBF5FF",
          "borderRadius": 4,
          "height": 24,
          "position": "absolute",
          "width": "0%",
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 11,
            "letterSpacing": 1.5,
            "lineHeight": 20,
          },
          Object {
            "color": "#4E5D78",
            "paddingLeft": 12,
            "zIndex": 1,
          },
        ]
      }
    >
      TODAY • 
    </Text>
  </View>
</View>
`;
